### ISSUE TITLE
Restoring only a certain folder or file

### ISSUE THREAD

**Issue by mholt** [restic/restic#1016]
This MIGHT just be me "holding it wrong" but I need some guidance any way. :smile:

## Output of `restic version`

```
restic 0.6.1 (v0.6.1-52-g5f449045)
compiled with go1.8.3 on linux/arm
```

## How did you start restic exactly? (Include the complete command line)

```
restic -r /mnt/bak/main restore latest --include /myfolder --target ~/restore
```

## What backend/server/service did you use?

Local file

## Expected behavior

Restore a specific folder within the root of the repo which is called `myfolder`. 

## Actual behavior

Didn't restore anything.

## Steps to reproduce the behavior

Do a backup with some folders in it, where two folders have the same name. Because after my initial command above didn't restore anything, I tried `--include myfolder/` but this restored BOTH folders named "myfolder" (the other one is deep in the dir tree) even though I really just want to restore the top-level one.

## Do you have any idea what may have caused this?

Maybe just confusion about how to use `--include`? I've read https://github.com/restic/restic/issues/396 and its linked issues, and a few other issues like https://github.com/restic/restic/issues/549.

Basically, just wondering about how to restore a specific folder only.

**Comment by fd0** [comment]
Thanks for the question, it seem that the manual needs extensive examples for include/exclude in restore/backup cases :smile:

You can do two things: If the directory name `myfolder` is unique in your backup, then you can just use `restic restore --include myfolder`. Or you can use an absolute include pattern, but this needs to be in the format which `restic ls` prints (with the strange directory structure, described in #549). If you ran `restic backup foo` and you'd like to restore `foo/myfolder`, then it's probably `restic restore --include /foo/myfolder --target /tmp/restore latest` or something similar.

**Comment by fd0** [comment]
There's a proposal to add code to optimize for this use case (restore a single dir/subdir from a snapshot), see #982.

**Comment by mholt** [comment]
Ah, indeed, using the absolute path as given by `restic ls` did the trick. The folder name is NOT unique in my case so I had to use that "repo-absolute" path.

I'm satisfied with this so I'll close the issue.
